src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts

```diff
@@ -509,33 +515,39 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 		this.badgeClassForm.controls.competencies.addFromTemplate();
 	}
 
-    /**
-     * Fetches competencies from the ai tool (@see aiSkillsService) and saves them
-     * in @see aiCompetenciesSuggestions. Also adds the necessary form control
-     * (@see badgeClassForm.controls.aiCompetencies) (and removes the old ones).
-     */
-    suggestCompetencies() {
-        if (this.aiCompetenciesDescription.length == 0) {
-            return;
-        }
-        this.aiSkillsService.getAiSkills(this.aiCompetenciesDescription)
-        .then(skills => {
-            this.aiCompetenciesSuggestions = skills;
-            let aiCompetencies = this.badgeClassForm.controls.aiCompetencies;
-            for (let i = aiCompetencies.length - 1; i >= 0; i--) {
-                aiCompetencies.removeAt(i);
-            }
-
-            skills.forEach(skill => {
-                aiCompetencies.addFromTemplate();
-            });
-        })
-        .catch(error => {
-            this.messageService.reportAndThrowError(
-                "Failed to obtain ai skills.",
-                error);
-        });
-    }
+	/**
+	 * Fetches competencies from the ai tool (@see aiSkillsService) and saves them
+	 * in @see aiCompetenciesSuggestions. Also adds the necessary form control
+	 * (@see badgeClassForm.controls.aiCompetencies) (and removes the old ones).
+	 */
+	suggestCompetencies() {
+		if (this.aiCompetenciesDescription.length == 0) {
+			return;
+		}
+		this.aiSkillsService
+			.getAiSkills(this.aiCompetenciesDescription)
+			.then((skills) => {
+				this.aiCompetenciesSuggestions = skills;
+				let aiCompetencies = this.badgeClassForm.controls.aiCompetencies;
+				console.log('aiCompetencies', aiCompetencies);
+				for (let i = aiCompetencies.length - 1; i >= 0; i--) {
+					aiCompetencies.removeAt(i);
+				}
+
+				skills.forEach((skill) => {
+					aiCompetencies.addFromTemplate();
+					this.showAiDetails.push(false);
+				});
+			})
+			.catch((error) => {
+				this.messageService.reportAndThrowError('Failed to obtain ai skills.', error);
+			});
+	}
+
+	toggleAiDetails(index: number): void {
+		console.log(this.badgeClassForm.controls.aiCompetencies.controls[index].rawControlMap.selected.value);
+		this.showAiDetails[index] = !this.showAiDetails[index];
```

Comment:
Isn't it possible to bind this value? E.g. via a form control thingy

Reply:
Not sure i understand, what would you bind this value to?

Reply:
Ah nevermind, I guess it wouldn't work because it's a button and not a checkbox